libs/blocks/chart/chart.js

```diff
@@ -48,20 +56,111 @@ async function fetchData(link) {
   return data;
 }
 
-export default async function init(el) {
-  const link = el.querySelector('a[href$="json"]');
-  if (!link) return;
+const barTooltipFormatter = ({
+  seriesName,
+  marker,
+  value,
+  encode: { x = [] },
+  name,
+} = {}, unit = '') => (
+  `${seriesName}<br />${marker} ${value[x[0]]}${unit} ${name}<i class="tooltip_icon"></i>`
+);
+
+const barSeriesOptions = (seriesData, colors, unit) => (
+  seriesData.map((value, index) => ({
+    type: 'bar',
+    label: {
+      show: true,
+      formatter: `{@[${index + 1}]}${unit}`,
+    },
+    showBackground: true,
+    backgroundStyle: { color: colors[index] },
+  }))
+);
+
+export const getChartOptions = (chartType, dataset, colors, unit = '') => {
+  const source = dataset?.source;
+  const seriesData = (source && source[1]) ? source[1].slice() : [];
+
+  seriesData.shift();
+
+  return {
+    dataset,
+    color: colors,
+    legend: {
+      show: true,
+      type: 'scroll',
+    },
+    tooltip: {
+      show: true,
+      formatter: (params) => barTooltipFormatter(params, unit),
+    },
+    xAxis: {
+      type: chartType === 'bar' ? 'value' : 'category',
```

Comment:
May want to create a `const isBarChart = chartType === 'bar'`
```suggestion
      type: isBarChart ? 'category' : 'value',
```

Reply:
The conditional will likely change as we add more chart types, so I'd like to defer this until those are added

Reply:
The conditional will likely change as we add more chart types, so I'd like to defer this until those are added